### DISCUSSION TITLE
"compared with upstream" diff-dwim options

### DISCUSSION THREAD

**Discussion by thejan2009** [magit/magit#4821] [Support] [answered]
Based on the discussion [here](https://github.com/magit/magit/issues/3851) and the [following commit](https://github.com/magit/magit/commit/9413847c1a085899d8de6f8d978bd7265f65e5d8), it seems the diff we get when pressing `RET` on `unpushed/unpulled to/from upstream` changed.

I was confused for some time (I guess a year or so) because the diff output was different than I remember and the previous behavior seemed much more useful. Anyway, the change can be reverted by overriding the following function:

```lisp
(defun magit-diff--range-to-endpoints (r) r)
```

I'm wondering if this is the best approach. (I also unuscessfully tried to `(fset magit-diff--range-to-endpoints 'identity)`)

**Comment by kyleam** [comment, marked as answer]
> I'm wondering if this is the best approach.

Rather than redefine it directly, I'd recommend overriding it with advice:

    (advice-add 'magit-diff--range-to-endpoints :override #'identity)
